`repeated interface: org.springframework.aop.SpringProxy` with spring-security annotations on `@RestController`

I get following error as soon as I use both `@RequiredArgsConstructor` and `@PreAuthorize` in the same  a `@RestController` :
```
[INFO]     [creator]     Error: Error parsing dynamic proxy configuration in file:/workspace/META-INF/native-image/org.springframework.aot/spring-aot/proxy-config.json:
[INFO]     [creator]     java.lang.IllegalArgumentException: repeated interface: org.springframework.aop.SpringProxy
[INFO]     [creator]     Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option DynamicProxyConfigurationResources.
[INFO]     [creator]     com.oracle.svm.core.util.UserError$UserException: Error parsing dynamic proxy configuration in file:/workspace/META-INF/native-image/org.springframework.aot/spring-aot/proxy-config.json:
[INFO]     [creator]     java.lang.IllegalArgumentException: repeated interface: org.springframework.aop.SpringProxy
[INFO]     [creator]     Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option DynamicProxyConfigurationResources.
[INFO]     [creator]            at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.doParseAndRegister(ConfigurationParserUtils.java:139)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.lambda$parseAndRegisterConfigurations$3(ConfigurationParserUtils.java:120)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:212)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils$1.tryAdvance(ConfigurationParserUtils.java:109)
[INFO]     [creator]            at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:326)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
[INFO]     [creator]            at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[INFO]     [creator]            at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
[INFO]     [creator]            at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[INFO]     [creator]            at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[INFO]     [creator]            at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:491)
[INFO]     [creator]            at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:449)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.parseAndRegisterConfigurations(ConfigurationParserUtils.java:126)
[INFO]     [creator]            at com.oracle.svm.reflect.proxy.hosted.DynamicProxyFeature.duringSetup(DynamicProxyFeature.java:80)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.lambda$setupNativeImage$18(NativeImageGenerator.java:915)
[INFO]     [creator]            at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:915)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:580)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$2(NativeImageGenerator.java:495)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[INFO]     [creator]            at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[INFO]     [creator]     Error: Image build request failed with exit status 1
[INFO]     [creator]     unable to invoke layer creator
[INFO]     [creator]     unable to contribute native-image layer
[INFO]     [creator]     error running build
[INFO]     [creator]     exit status 1
[INFO]     [creator]     ERROR: failed to build: exit status 1
```

This can be reproduced starting from `security-method-webflux` sample and running `mvn spring-boot:build-image`. A PR reproducing the error follows.

I formatted my hard-drive and installed both Windows 10 and Fedora 34. With fresh install and exact same code, both build fail with different errors.

I've updated spring-boot to 2.5.3, but kept spring native 0.10.1 as I have errors with current 0.10.2-SNAPSHOT even on webmvc module

``` bash
git clone git@github.com:ch4mpy/starter.git
cd starter/api
mvn clean install -DskipTests
mvn package -Pbuild-image -DskipTests
```
- on Windows, as before:
```
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:211]        setup:   1,862.84 ms,  1.67 GB
[INFO]     [creator]     Error: Error parsing dynamic proxy configuration in file:/workspace/META-INF/native-image/org.springframework.aot/spring-aot/proxy-config.json:
[INFO]     [creator]     java.lang.IllegalArgumentException: repeated interface: org.springframework.aop.SpringProxy
[INFO]     [creator]     Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option DynamicProxyConfigurationResources.
[INFO]     [creator]     com.oracle.svm.core.util.UserError$UserException: Error parsing dynamic proxy configuration in file:/workspace/META-INF/native-image/org.springframework.ot/spring-aot/proxy-config.json:
[INFO]     [creator]     java.lang.IllegalArgumentException: repeated interface: org.springframework.aop.SpringProxy
[INFO]     [creator]     Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option DynamicProxyConfigurationResources.
[INFO]     [creator]            at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.doParseAndRegister(ConfigurationParserUtils.java:133)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.lambda$parseAndRegisterConfigurations$3(ConfigurationParserUtils.java:111)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:212)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils$1.tryAdvance(ConfigurationParserUtils.java:104)
[INFO]     [creator]            at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:326)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
[INFO]     [creator]            at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
[INFO]     [creator]            at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[INFO]     [creator]            at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
[INFO]     [creator]            at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[INFO]     [creator]            at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
[INFO]     [creator]            at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[INFO]     [creator]            at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:491)
[INFO]     [creator]            at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:449)
[INFO]     [creator]            at com.oracle.svm.hosted.config.ConfigurationParserUtils.parseAndRegisterConfigurations(ConfigurationParserUtils.java:113)
[INFO]     [creator]            at com.oracle.svm.reflect.proxy.hosted.DynamicProxyFeature.duringSetup(DynamicProxyFeature.java:81)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.lambda$setupNativeImage$16(NativeImageGenerator.java:896)
[INFO]     [creator]            at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:896)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:530)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:491)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:380)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:543)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:119)
[INFO]     [creator]            at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:573)
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:211]      [total]:   5,211.31 ms,  1.67 GB
[INFO]     [creator]     # Printing build artifacts to: /layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication.build_artifacts.txt
[INFO]     [creator]     Error: Image build request failed with exit status 1
[INFO]     [creator]     unable to invoke layer creator
[INFO]     [creator]     unable to contribute native-image layer
[INFO]     [creator]     error running build
[INFO]     [creator]     exit status 1
[INFO]     [creator]     ERROR: failed to build: exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for starter-parent 0.0.1-SNAPSHOT:
[INFO]
[INFO] starter-parent ..................................... SUCCESS [  0.152 s]
[INFO] starter-parent-webflux ............................. SUCCESS [  0.000 s]
[INFO] common-exception-webflux ........................... SUCCESS [  1.621 s]
[INFO] common-security-webflux ............................ SUCCESS [  0.751 s]
[INFO] starter-domain-relational .......................... SUCCESS [  0.836 s]
[INFO] common-storage ..................................... SUCCESS [  0.611 s]
[INFO] starter-api-webflux ................................ FAILURE [04:34 min]
[INFO] starter-parent-webmvc .............................. SKIPPED
[INFO] common-exception-webmvc ............................ SKIPPED
[INFO] common-security-webmvc ............................. SKIPPED
[INFO] starter-domain-jpa ................................. SKIPPED
[INFO] starter-api-webmvc ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:39 min
[INFO] Finished at: 2021-08-06T13:45:52-10:00
[INFO] ------------------------------------------------------------------------
```
- on Linux:
```
c4_soft.starter.FaultsEndpointsApplication:203]     (clinit):   1,993.66 ms,  8.00 GB
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:203]   (typeflow):  89,390.58 ms,  8.00 GB
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:203]    (objects):  52,017.21 ms,  8.00 GB
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:203]   (features):  10,484.74 ms,  8.00 GB
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:203]     analysis: 158,192.88 ms,  8.00 GB
[INFO]     [creator]     Error: Class initialization of io.netty.handler.codec.compression.BrotliDecoder failed. Use the option --initialize-at-run-time=io.netty.handler.codec.compression.BrotliDecoder to explicitly request delayed initialization of this class.
[INFO]     [creator]     Detailed message:
[INFO]     [creator]     
[INFO]     [creator]     com.oracle.svm.core.util.UserError$UserException: Class initialization of io.netty.handler.codec.compression.BrotliDecoder failed. Use the option --initialize-at-run-time=io.netty.handler.codec.compression.BrotliDecoder to explicitly request delayed initialization of this class.
[INFO]     [creator]     Detailed message:
[INFO]     [creator]     
[INFO]     [creator]     	at com.oracle.svm.core.util.UserError.abort(UserError.java:82)
[INFO]     [creator]     	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:764)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:532)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:491)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:380)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:543)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:119)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:573)
[INFO]     [creator]     Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Class initialization of io.netty.handler.codec.compression.BrotliDecoder failed. Use the option --initialize-at-run-time=io.netty.handler.codec.compression.BrotliDecoder to explicitly request delayed initialization of this class.
[INFO]     [creator]     Detailed message:
[INFO]     [creator]     
[INFO]     [creator]     	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:761)
[INFO]     [creator]     	... 6 more
[INFO]     [creator]     Caused by: java.lang.ExceptionInInitializerError
[INFO]     [creator]     	at io.netty.handler.codec.compression.BrotliDecoder.<clinit>(BrotliDecoder.java:44)
[INFO]     [creator]     	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
[INFO]     [creator]     	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1042)
[INFO]     [creator]     	at jdk.unsupported/sun.misc.Unsafe.ensureClassInitialized(Unsafe.java:698)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.serviceprovider.GraalUnsafeAccess.ensureClassInitialized(GraalUnsafeAccess.java:77)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.ensureClassInitialized(ConfigurableClassInitialization.java:178)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:648)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:131)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:159)
[INFO]     [creator]     	at com.oracle.svm.hosted.SVMHost.isInitialized(SVMHost.java:307)
[INFO]     [creator]     	at com.oracle.graal.pointsto.meta.AnalysisType.isInitialized(AnalysisType.java:819)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.maybeEagerlyInitialize(BytecodeParser.java:4389)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genNewInstance(BytecodeParser.java:4587)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genNewInstance(BytecodeParser.java:4576)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genNewInstance(BytecodeParser.java:4571)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5376)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3431)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3391)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3236)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1122)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1007)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
[INFO]     [creator]     	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:81)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:212)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:131)
[INFO]     [creator]     	at com.oracle.svm.hosted.SVMHost.parseBytecode(SVMHost.java:709)
[INFO]     [creator]     	at com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:605)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:163)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:321)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.MethodTypeFlow.createTypeFlow(MethodTypeFlow.java:293)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureTypeFlowCreated(MethodTypeFlow.java:282)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:103)
[INFO]     [creator]     	at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:222)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.TypeFlow.notifyObservers(TypeFlow.java:471)
[INFO]     [creator]     	at com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:540)
[INFO]     [creator]     	at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:539)
[INFO]     [creator]     	at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:188)
[INFO]     [creator]     	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:172)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[INFO]     [creator]     	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[INFO]     [creator]     Caused by: java.lang.ClassNotFoundException: com.aayushatharva.brotli4j.Brotli4jLoader
[INFO]     [creator]     	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
[INFO]     [creator]     	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
[INFO]     [creator]     	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
[INFO]     [creator]     	at java.base/java.lang.Class.forName0(Native Method)
[INFO]     [creator]     	at java.base/java.lang.Class.forName(Class.java:398)
[INFO]     [creator]     	at io.netty.handler.codec.compression.Brotli.<clinit>(Brotli.java:33)
[INFO]     [creator]     	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
[INFO]     [creator]     	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1042)
[INFO]     [creator]     	at jdk.unsupported/sun.misc.Unsafe.ensureClassInitialized(Unsafe.java:698)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.serviceprovider.GraalUnsafeAccess.ensureClassInitialized(GraalUnsafeAccess.java:77)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.ensureClassInitialized(ConfigurableClassInitialization.java:178)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:648)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:131)
[INFO]     [creator]     	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:159)
[INFO]     [creator]     	at com.oracle.svm.hosted.SVMHost.isInitialized(SVMHost.java:307)
[INFO]     [creator]     	at com.oracle.graal.pointsto.meta.AnalysisType.isInitialized(AnalysisType.java:819)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.maybeEagerlyInitialize(BytecodeParser.java:4389)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeStatic(BytecodeParser.java:1636)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeStatic(BytecodeParser.java:1629)
[INFO]     [creator]     	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5373)
[INFO]     [creator]     	... 31 more
[INFO]     [creator]     [/layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication:203]      [total]: 165,949.74 ms,  8.00 GB
[INFO]     [creator]     # Printing build artifacts to: /layers/paketo-buildpacks_native-image/native-image/com.c4_soft.starter.FaultsEndpointsApplication.build_artifacts.txt
[INFO]     [creator]     Error: Image build request failed with exit status 1
[INFO]     [creator]     unable to invoke layer creator
[INFO]     [creator]     unable to contribute native-image layer
[INFO]     [creator]     error running build
[INFO]     [creator]     exit status 1
[INFO]     [creator]     ERROR: failed to build: exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for starter-parent 0.0.1-SNAPSHOT:
[INFO] 
[INFO] starter-parent ..................................... SUCCESS [  0.197 s]
[INFO] starter-parent-webflux ............................. SUCCESS [  0.003 s]
[INFO] common-exception-webflux ........................... SUCCESS [  1.895 s]
[INFO] common-security-webflux ............................ SUCCESS [  0.772 s]
[INFO] starter-domain-relational .......................... SUCCESS [  0.849 s]
[INFO] common-storage ..................................... SUCCESS [  0.504 s]
[INFO] starter-api-webflux ................................ FAILURE [05:10 min]
[INFO] starter-parent-webmvc .............................. SKIPPED
[INFO] common-exception-webmvc ............................ SKIPPED
[INFO] common-security-webmvc ............................. SKIPPED
[INFO] starter-domain-jpa ................................. SKIPPED
[INFO] starter-api-webmvc ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:15 min
[INFO] Finished at: 2021-08-06T13:00:04-10:00
[INFO] ------------------------------------------------------------------------
```

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/AotProxyDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/AotProxyDescriptor.java
@@ -18,7 +18,7 @@ package org.springframework.nativex.domain.proxies;
 
 import java.util.Arrays;
 import java.util.Collection;
-import java.util.SortedSet;
+import java.util.Set;
 
 import org.springframework.aop.framework.BuildTimeProxyDescriptor;
 import org.springframework.nativex.hint.ProxyBits;
@@ -44,11 +44,11 @@ public class AotProxyDescriptor extends JdkProxyDescriptor {
 		return targetClassName;
 	}
 
-	public SortedSet<String> getInterfaceTypes() {
+	public Set<String> getInterfaceTypes() {
 		return types;
 	}
 
-	public SortedSet<String> getTypes() {
+	public Set<String> getTypes() {
 		throw new IllegalStateException();
 	}
 	

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/AotProxyDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/AotProxyDescriptor.java
@@ -17,7 +17,8 @@
 package org.springframework.nativex.domain.proxies;
 
 import java.util.Arrays;
-import java.util.List;
+import java.util.Collection;
+import java.util.SortedSet;
 
 import org.springframework.aop.framework.BuildTimeProxyDescriptor;
 import org.springframework.nativex.hint.ProxyBits;
@@ -25,6 +26,7 @@ import org.springframework.nativex.hint.ProxyBits;
 /**
  * 
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class AotProxyDescriptor extends JdkProxyDescriptor {
 	
@@ -32,7 +34,7 @@ public class AotProxyDescriptor extends JdkProxyDescriptor {
 
 	private final int proxyFeatures;
 	
-	public AotProxyDescriptor(String targetClassName, List<String> interfaceNames, int proxyFeatures) {
+	public AotProxyDescriptor(String targetClassName, Collection<String> interfaceNames, int proxyFeatures) {
 		super(interfaceNames);
 		this.targetClassName = targetClassName;
 		this.proxyFeatures = proxyFeatures;
@@ -42,11 +44,11 @@ public class AotProxyDescriptor extends JdkProxyDescriptor {
 		return targetClassName;
 	}
 
-	public List<String> getInterfaceTypes() {
+	public SortedSet<String> getInterfaceTypes() {
 		return types;
 	}
 
-	public List<String> getTypes() {
+	public SortedSet<String> getTypes() {
 		throw new IllegalStateException();
 	}
 	

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/JdkProxyDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/JdkProxyDescriptor.java
@@ -17,8 +17,9 @@
 package org.springframework.nativex.domain.proxies;
 
 import java.util.Collection;
-import java.util.SortedSet;
-import java.util.TreeSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
 
 /**
  * Describes a proxy via a set of types it should implement.
@@ -28,13 +29,13 @@ import java.util.TreeSet;
  */
 public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 
-	protected SortedSet<String> types; // e.g. java.io.Serializable
+	protected Set<String> types; // e.g. java.io.Serializable
 
 	JdkProxyDescriptor() {
 	}
 
 	public JdkProxyDescriptor(Collection<String> types) {
-		this.types = new TreeSet<>(types);
+		this.types = new LinkedHashSet<>(types);
 	}
 
 	@Override
@@ -86,12 +87,21 @@ public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 
 	@Override
 	public int compareTo(JdkProxyDescriptor o) {
-		SortedSet<String> l = this.types;
-		SortedSet<String> r = o.types;
+		Set<String> l = this.types;
+		Set<String> r = o.types;
 		if (l.size() != r.size()) {
 			return l.size() - r.size();
 		}
-		return l.containsAll(r) ? 0 : 1; // equal!
+		if (l.isEmpty()) return 0;
+		
+		Iterator<String> iterator1 = l.iterator();
+		Iterator<String> iterator2 = r.iterator();
+		while (iterator1.hasNext() && iterator2.hasNext()) {
+			int value = iterator1.next().compareTo(iterator2.next());
+			if (value != 0)
+				return value;
+		}
+		return 0; // equal!
 	}
 
 	public static JdkProxyDescriptor of(Collection<String> interfaces) {
@@ -101,15 +111,14 @@ public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 	}
 
 	public void setInterfaces(Collection<String> interfaces) {
-		this.types = new TreeSet<>();
-		this.types.addAll(interfaces);
+		this.types = new LinkedHashSet<>(interfaces);
 	}
 
 	public boolean containsInterface(String intface) {
 		return types.contains(intface);
 	}
 
-	public SortedSet<String> getTypes() {
+	public Set<String> getTypes() {
 		return types;
 	}
 

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/JdkProxyDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/JdkProxyDescriptor.java
@@ -16,23 +16,25 @@
 
 package org.springframework.nativex.domain.proxies;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Collection;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 /**
  * Describes a proxy via a set of types it should implement.
  *
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 
-	protected List<String> types; // e.g. java.io.Serializable
+	protected SortedSet<String> types; // e.g. java.io.Serializable
 
 	JdkProxyDescriptor() {
 	}
 
-	public JdkProxyDescriptor(List<String> types) {
-		this.types = new ArrayList<>(types);
+	public JdkProxyDescriptor(Collection<String> types) {
+		this.types = new TreeSet<>(types);
 	}
 
 	@Override
@@ -84,28 +86,22 @@ public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 
 	@Override
 	public int compareTo(JdkProxyDescriptor o) {
-		List<String> l = this.types;
-		List<String> r = o.types;
+		SortedSet<String> l = this.types;
+		SortedSet<String> r = o.types;
 		if (l.size() != r.size()) {
 			return l.size() - r.size();
 		}
-		for (int i = 0; i < l.size(); i++) {
-			int cmpTo = l.get(i).compareTo(r.get(i));
-			if (cmpTo != 0) {
-				return cmpTo;
-			}
-		}
-		return 0; // equal!
+		return l.containsAll(r) ? 0 : 1; // equal!
 	}
 
-	public static JdkProxyDescriptor of(List<String> interfaces) {
+	public static JdkProxyDescriptor of(Collection<String> interfaces) {
 		JdkProxyDescriptor pd = new JdkProxyDescriptor();
 		pd.setInterfaces(interfaces);
 		return pd;
 	}
 
-	public void setInterfaces(List<String> interfaces) {
-		this.types = new ArrayList<>();
+	public void setInterfaces(Collection<String> interfaces) {
+		this.types = new TreeSet<>();
 		this.types.addAll(interfaces);
 	}
 
@@ -113,7 +109,7 @@ public class JdkProxyDescriptor implements Comparable<JdkProxyDescriptor> {
 		return types.contains(intface);
 	}
 
-	public List<String> getTypes() {
+	public SortedSet<String> getTypes() {
 		return types;
 	}
 

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptor.java
@@ -16,28 +16,30 @@
 
 package org.springframework.nativex.domain.proxies;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Collection;
+import java.util.SortedSet;
+import java.util.TreeSet;
 import java.util.function.Consumer;
 
 /**
  * https://github.com/oracle/graal/blob/master/substratevm/REFLECTION.md
  * 
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class ProxiesDescriptor {
 
-	private final List<JdkProxyDescriptor> proxyDescriptors;
+	private final SortedSet<JdkProxyDescriptor> proxyDescriptors;
 
 	public ProxiesDescriptor() {
-		this.proxyDescriptors = new ArrayList<>();
+		this.proxyDescriptors = new TreeSet<>();
 	}
 
 	public ProxiesDescriptor(ProxiesDescriptor metadata) {
-		this.proxyDescriptors = new ArrayList<>(metadata.proxyDescriptors);
+		this.proxyDescriptors = new TreeSet<>(metadata.proxyDescriptors);
 	}
 
-	public List<JdkProxyDescriptor> getProxyDescriptors() {
+	public SortedSet<JdkProxyDescriptor> getProxyDescriptors() {
 		return this.proxyDescriptors;
 	}
 
@@ -75,7 +77,7 @@ public class ProxiesDescriptor {
 		}
 	}
 
-	public void consume(Consumer<List<String>> consumer) {
+	public void consume(Consumer<Collection<String>> consumer) {
 		proxyDescriptors.stream().forEach(pd -> consumer.accept(pd.getTypes()));
 	}
 

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptor.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptor.java
@@ -17,8 +17,8 @@
 package org.springframework.nativex.domain.proxies;
 
 import java.util.Collection;
-import java.util.SortedSet;
-import java.util.TreeSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
 import java.util.function.Consumer;
 
 /**
@@ -29,17 +29,17 @@ import java.util.function.Consumer;
  */
 public class ProxiesDescriptor {
 
-	private final SortedSet<JdkProxyDescriptor> proxyDescriptors;
+	private final Set<JdkProxyDescriptor> proxyDescriptors;
 
 	public ProxiesDescriptor() {
-		this.proxyDescriptors = new TreeSet<>();
+		this.proxyDescriptors = new LinkedHashSet<>();
 	}
 
 	public ProxiesDescriptor(ProxiesDescriptor metadata) {
-		this.proxyDescriptors = new TreeSet<>(metadata.proxyDescriptors);
+		this.proxyDescriptors = new LinkedHashSet<>(metadata.proxyDescriptors);
 	}
 
-	public SortedSet<JdkProxyDescriptor> getProxyDescriptors() {
+	public Set<JdkProxyDescriptor> getProxyDescriptors() {
 		return this.proxyDescriptors;
 	}
 

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonConverter.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonConverter.java
@@ -16,7 +16,7 @@
 
 package org.springframework.nativex.domain.proxies;
 
-import java.util.SortedSet;
+import java.util.Set;
 
 import org.springframework.nativex.json.JSONArray;
 
@@ -38,7 +38,7 @@ class ProxiesDescriptorJsonConverter {
 
 	public JSONArray toJsonArray(JdkProxyDescriptor pd) throws Exception {
 		JSONArray jsonArray = new JSONArray();
-		SortedSet<String> interfaces = pd.getTypes();
+		Set<String> interfaces = pd.getTypes();
 		for (String intface: interfaces) {
 			jsonArray.put(intface);
 		}

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonConverter.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonConverter.java
@@ -16,7 +16,7 @@
 
 package org.springframework.nativex.domain.proxies;
 
-import java.util.List;
+import java.util.SortedSet;
 
 import org.springframework.nativex.json.JSONArray;
 
@@ -24,6 +24,7 @@ import org.springframework.nativex.json.JSONArray;
  * Converter to change reflection descriptor objects into JSON objects
  *
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 class ProxiesDescriptorJsonConverter {
 
@@ -37,7 +38,7 @@ class ProxiesDescriptorJsonConverter {
 
 	public JSONArray toJsonArray(JdkProxyDescriptor pd) throws Exception {
 		JSONArray jsonArray = new JSONArray();
-		List<String> interfaces = pd.getTypes();
+		SortedSet<String> interfaces = pd.getTypes();
 		for (String intface: interfaces) {
 			jsonArray.put(intface);
 		}

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonMarshaller.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonMarshaller.java
@@ -23,8 +23,8 @@ import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.springframework.nativex.json.JSONArray;
 
@@ -32,6 +32,7 @@ import org.springframework.nativex.json.JSONArray;
  * Marshaller to write {@link ProxiesDescriptor} as JSON.
  *
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class ProxiesDescriptorJsonMarshaller {
 
@@ -90,7 +91,7 @@ public class ProxiesDescriptorJsonMarshaller {
 	
 	private static JdkProxyDescriptor toProxyDescriptor(JSONArray array) throws Exception {
 		JdkProxyDescriptor pd = new JdkProxyDescriptor();
-		List<String> interfaces = new ArrayList<>();
+		Set<String> interfaces = new HashSet<>();
 		for (int i=0;i<array.length();i++) {
 			interfaces.add(array.getString(i));
 		}

--- a/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonMarshaller.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/domain/proxies/ProxiesDescriptorJsonMarshaller.java
@@ -23,8 +23,8 @@ import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.ArrayList;
+import java.util.Collection;
 
 import org.springframework.nativex.json.JSONArray;
 
@@ -91,7 +91,7 @@ public class ProxiesDescriptorJsonMarshaller {
 	
 	private static JdkProxyDescriptor toProxyDescriptor(JSONArray array) throws Exception {
 		JdkProxyDescriptor pd = new JdkProxyDescriptor();
-		Set<String> interfaces = new HashSet<>();
+		Collection<String> interfaces = new ArrayList<>();
 		for (int i=0;i<array.length();i++) {
 			interfaces.add(array.getString(i));
 		}

--- a/spring-aot/src/main/java/org/springframework/nativex/support/ConfigurationCollector.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/support/ConfigurationCollector.java
@@ -54,6 +54,7 @@ import org.springframework.nativex.type.TypeSystem;
  * Centralized collector of all computed configuration, can be used to produce JSON files.
  * 
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class ConfigurationCollector {
 
@@ -146,9 +147,8 @@ public class ConfigurationCollector {
 		this.ts = ts;
 	}
 
-	private boolean checkTypes(List<String> types, Predicate<Type> test) {
-		for (int i = 0; i < types.size(); i++) {
-			String className = types.get(i);
+	private boolean checkTypes(Collection<String> types, Predicate<Type> test) {
+		for (String className : types) {
 			Type clazz = ts.resolveDotted(className, true);
 			if (!test.test(clazz)) {
 				return false;
@@ -157,7 +157,7 @@ public class ConfigurationCollector {
 		return true;
 	}
 
-	public boolean addProxy(List<String> interfaceNames, boolean verify) {
+	public boolean addProxy(Collection<String> interfaceNames, boolean verify) {
 		if (verify) {
 			if (!checkTypes(interfaceNames, t -> t!=null && t.isInterface())) {
 				return false;

--- a/spring-aot/src/main/java/org/springframework/nativex/support/DynamicProxiesHandler.java
+++ b/spring-aot/src/main/java/org/springframework/nativex/support/DynamicProxiesHandler.java
@@ -16,7 +16,7 @@
 
 package org.springframework.nativex.support;
 
-import java.util.List;
+import java.util.Collection;
 
 import org.springframework.nativex.domain.proxies.AotProxyDescriptor;
 import org.springframework.nativex.domain.proxies.JdkProxyDescriptor;
@@ -24,6 +24,7 @@ import org.springframework.nativex.domain.proxies.JdkProxyDescriptor;
 /**
  * 
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class DynamicProxiesHandler extends Handler {
 
@@ -39,12 +40,12 @@ public class DynamicProxiesHandler extends Handler {
 		}
 	}
 	
-	public boolean addClassProxy(String targetClassName, List<String> interfaceNames, int proxyFeatures) {
+	public boolean addClassProxy(String targetClassName, Collection<String> interfaceNames, int proxyFeatures) {
 		AotProxyDescriptor cpd = new AotProxyDescriptor(targetClassName, interfaceNames, proxyFeatures);
 		return collector.addClassProxy(cpd, true);
 	}
 
-	public boolean addProxy(List<String> interfaceNames) {
+	public boolean addProxy(Collection<String> interfaceNames) {
 		return collector.addProxy(interfaceNames, true);
 	}
 }

--- a/spring-aot/src/test/java/org/springframework/nativex/DescriptorTests.java
+++ b/spring-aot/src/test/java/org/springframework/nativex/DescriptorTests.java
@@ -90,8 +90,8 @@ public class DescriptorTests {
 		ProxiesDescriptor pd2 = ProxiesDescriptor.fromJSON(json);
 		assertThat(pd.toString()).isEqualTo(pd2.toString());
 		assertThat(d.containsInterface("java.io.Serializable")).isTrue();
-		assertThat(d.compareTo(pd2.getProxyDescriptors().get(0))).isEqualTo(0);
-		assertThat(d.equals(pd2.getProxyDescriptors().get(0))).isTrue();
+		assertThat(d.compareTo(pd2.getProxyDescriptors().first())).isEqualTo(0);
+		assertThat(d.equals(pd2.getProxyDescriptors().first())).isTrue();
 	}
 
 	@Test

--- a/spring-aot/src/test/java/org/springframework/nativex/DescriptorTests.java
+++ b/spring-aot/src/test/java/org/springframework/nativex/DescriptorTests.java
@@ -90,8 +90,8 @@ public class DescriptorTests {
 		ProxiesDescriptor pd2 = ProxiesDescriptor.fromJSON(json);
 		assertThat(pd.toString()).isEqualTo(pd2.toString());
 		assertThat(d.containsInterface("java.io.Serializable")).isTrue();
-		assertThat(d.compareTo(pd2.getProxyDescriptors().first())).isEqualTo(0);
-		assertThat(d.equals(pd2.getProxyDescriptors().first())).isTrue();
+		assertThat(d.compareTo(pd2.getProxyDescriptors().iterator().next())).isEqualTo(0);
+		assertThat(d.equals(pd2.getProxyDescriptors().iterator().next())).isTrue();
 	}
 
 	@Test

--- a/spring-aot/src/test/java/org/springframework/nativex/HintTests.java
+++ b/spring-aot/src/test/java/org/springframework/nativex/HintTests.java
@@ -115,8 +115,8 @@ public class HintTests {
 		List<JdkProxyDescriptor> proxies= hint.getProxyDescriptors();
 		assertEquals(1,proxies.size());
 		String[] types = proxies.get(0).getTypes().toArray(new String[] {}); 
-		assertEquals("java.lang.Integer",types[0]);
-		assertEquals("java.lang.String",types[1]);
+		assertEquals("java.lang.String",types[0]);
+		assertEquals("java.lang.Integer",types[1]);
 	}
 
 	@NativeHint(jdkProxies = { @JdkProxyHint(types = { String.class,Integer.class }) })
@@ -260,7 +260,7 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.String");
-		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.io.Serializable");
+		assertThat(cpd.getInterfaceTypes().iterator().next()).isEqualTo("java.io.Serializable");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 
@@ -281,7 +281,7 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.String");
-		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.util.List");
+		assertThat(cpd.getInterfaceTypes().iterator().next()).isEqualTo("java.util.List");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 
@@ -312,12 +312,12 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.Integer");
-		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.util.List");
+		assertThat(cpd.getInterfaceTypes().iterator().next()).isEqualTo("java.util.List");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd2 = (AotProxyDescriptor)proxyDescriptors.get(1);
 		assertThat(cpd2.getTargetClassType()).isEqualTo("java.lang.Number");
-		assertThat(cpd2.getInterfaceTypes().first()).isEqualTo("java.util.List");
+		assertThat(cpd2.getInterfaceTypes().iterator().next()).isEqualTo("java.util.List");
 		assertThat(cpd2.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 	

--- a/spring-aot/src/test/java/org/springframework/nativex/HintTests.java
+++ b/spring-aot/src/test/java/org/springframework/nativex/HintTests.java
@@ -115,8 +115,8 @@ public class HintTests {
 		List<JdkProxyDescriptor> proxies= hint.getProxyDescriptors();
 		assertEquals(1,proxies.size());
 		String[] types = proxies.get(0).getTypes().toArray(new String[] {}); 
-		assertEquals("java.lang.String",types[0]);
-		assertEquals("java.lang.Integer",types[1]);
+		assertEquals("java.lang.Integer",types[0]);
+		assertEquals("java.lang.String",types[1]);
 	}
 
 	@NativeHint(jdkProxies = { @JdkProxyHint(types = { String.class,Integer.class }) })
@@ -260,7 +260,7 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.String");
-		assertThat(cpd.getInterfaceTypes().get(0)).isEqualTo("java.io.Serializable");
+		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.io.Serializable");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 
@@ -281,7 +281,7 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.String");
-		assertThat(cpd.getInterfaceTypes().get(0)).isEqualTo("java.util.List");
+		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.util.List");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 
@@ -312,12 +312,12 @@ public class HintTests {
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd = (AotProxyDescriptor)proxyDescriptors.get(0);
 		assertThat(cpd.getTargetClassType()).isEqualTo("java.lang.Integer");
-		assertThat(cpd.getInterfaceTypes().get(0)).isEqualTo("java.util.List");
+		assertThat(cpd.getInterfaceTypes().first()).isEqualTo("java.util.List");
 		assertThat(cpd.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 		assertThat(proxyDescriptors.get(0).isClassProxy()).isTrue();
 		AotProxyDescriptor cpd2 = (AotProxyDescriptor)proxyDescriptors.get(1);
 		assertThat(cpd2.getTargetClassType()).isEqualTo("java.lang.Number");
-		assertThat(cpd2.getInterfaceTypes().get(0)).isEqualTo("java.util.List");
+		assertThat(cpd2.getInterfaceTypes().first()).isEqualTo("java.util.List");
 		assertThat(cpd2.getProxyFeatures()).isEqualTo(ProxyBits.EXPOSE_PROXY);
 	}
 	

--- a/spring-native-configuration/src/test/java/org/springframework/nativex/util/HintDeclarationAssert.java
+++ b/spring-native-configuration/src/test/java/org/springframework/nativex/util/HintDeclarationAssert.java
@@ -56,7 +56,7 @@ public class HintDeclarationAssert extends AbstractAssert<HintDeclarationAssert,
 	public HintDeclarationAssert containsProxyFor(Class<?> type) {
 
 		if (!actual.getProxyDescriptors().stream().map(JdkProxyDescriptor::getTypes)
-				.anyMatch(types -> type.getName().equals(types.get(0)))) {
+				.anyMatch(types -> type.getName().equals(types.first()))) {
 
 			failWithMessage("Expected HintDeclaration to contain proxy config for %s but was not in %s",
 					type.getName(), actual.getProxyDescriptors());

--- a/spring-native-configuration/src/test/java/org/springframework/nativex/util/HintDeclarationAssert.java
+++ b/spring-native-configuration/src/test/java/org/springframework/nativex/util/HintDeclarationAssert.java
@@ -56,7 +56,7 @@ public class HintDeclarationAssert extends AbstractAssert<HintDeclarationAssert,
 	public HintDeclarationAssert containsProxyFor(Class<?> type) {
 
 		if (!actual.getProxyDescriptors().stream().map(JdkProxyDescriptor::getTypes)
-				.anyMatch(types -> type.getName().equals(types.first()))) {
+				.anyMatch(types -> type.getName().equals(types.iterator().next()))) {
 
 			failWithMessage("Expected HintDeclaration to contain proxy config for %s but was not in %s",
 					type.getName(), actual.getProxyDescriptors());

--- a/spring-native/src/main/java/org/springframework/aop/framework/BuildTimeProxyDescriptor.java
+++ b/spring-native/src/main/java/org/springframework/aop/framework/BuildTimeProxyDescriptor.java
@@ -17,8 +17,9 @@
 package org.springframework.aop.framework;
 
 import java.util.Collection;
-import java.util.SortedSet;
-import java.util.TreeSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
 
 /**
  * 
@@ -29,13 +30,13 @@ public class BuildTimeProxyDescriptor {
 
 	private final String targetClassName;
 	
-	private final SortedSet<String> interfaceNames;
+	private final Set<String> interfaceNames;
 
 	private final int proxyFeatures; // ProxyBits
 	
 	public BuildTimeProxyDescriptor(String targetClassName, Collection<String> interfaceNames, int proxyFeatures) {
 		this.targetClassName = targetClassName;
-		this.interfaceNames = new TreeSet<String>(interfaceNames);
+		this.interfaceNames = new LinkedHashSet<String>(interfaceNames);
 		this.proxyFeatures = proxyFeatures;
 	}
 
@@ -43,7 +44,7 @@ public class BuildTimeProxyDescriptor {
 		return (proxyFeatures&bit)!=0;
 	}
 
-	public SortedSet<String> getInterfaceTypes() {
+	public Set<String> getInterfaceTypes() {
 		return interfaceNames;
 	}
 

--- a/spring-native/src/main/java/org/springframework/aop/framework/BuildTimeProxyDescriptor.java
+++ b/spring-native/src/main/java/org/springframework/aop/framework/BuildTimeProxyDescriptor.java
@@ -16,23 +16,26 @@
 
 package org.springframework.aop.framework;
 
-import java.util.List;
+import java.util.Collection;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 /**
  * 
  * @author Andy Clement
+ * @author Ariel Carrera
  */
 public class BuildTimeProxyDescriptor {
 
 	private final String targetClassName;
 	
-	private final List<String> interfaceNames;
+	private final SortedSet<String> interfaceNames;
 
 	private final int proxyFeatures; // ProxyBits
 	
-	public BuildTimeProxyDescriptor(String targetClassName, List<String> interfaceNames, int proxyFeatures) {
+	public BuildTimeProxyDescriptor(String targetClassName, Collection<String> interfaceNames, int proxyFeatures) {
 		this.targetClassName = targetClassName;
-		this.interfaceNames = interfaceNames;
+		this.interfaceNames = new TreeSet<String>(interfaceNames);
 		this.proxyFeatures = proxyFeatures;
 	}
 
@@ -40,7 +43,7 @@ public class BuildTimeProxyDescriptor {
 		return (proxyFeatures&bit)!=0;
 	}
 
-	public List<String> getInterfaceTypes() {
+	public SortedSet<String> getInterfaceTypes() {
 		return interfaceNames;
 	}
 

